src/test/java/mate/academy/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceImplTest {
+    private static final long USER_ID = 1L;
+    private static final String USER_EMAIL = "john@me.com";
+    private static final String USER_PASSWORD = "12345678";
+    private static final long NON_EXIST_ID = 7L;
+    private static final String NON_EXIST_EMAIL = "mary@me.com";
+    private User user;
+    private UserDao userDao;
+    private UserService userService;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(USER_ID);
+        user.setEmail(USER_EMAIL);
+        user.setPassword(USER_PASSWORD);
+        user.setRoles(Set.of(new Role(Role.RoleName.ADMIN)));
```

Comment:
Create a constant for the Role instance to avoid creating a new Role object in each test.